lib/Dialect/TorchConversion/Transforms/VerifyTosaLinalgBackendContract.cpp

```diff
@@ -39,12 +41,19 @@ class VerifyTosaLinalgBackendContractPass
   void runOnOperation() override {
     MLIRContext *context = &getContext();
     auto module = getOperation();
+
     TypeConverter converter;
-    converter.addConversion([](RankedTensorType type) -> Type {
-      if (BaseMemRefType::isValidElementType(type.getElementType()))
+    converter.addConversion([](Type type) -> Type {
+      auto elemTy = type;
+      if (isa<TensorType>(type))
+        elemTy = cast<TensorType>(type).getElementType();
+      if (isa<MemRefType>(type))
+        elemTy = cast<MemRefType>(type).getElementType();      
+      if (BaseMemRefType::isValidElementType(elemTy))
```

Comment:
Why was this change necessary?

Reply:
This is needed because during the verification of the legalization of `MemRefTypes,` we would fail while getting the element type without this change